src/subgraphs/getData.ts

```diff
@@ -83,31 +86,50 @@ export class DataRecoveryGraph extends AbstractGraph<DataRecoveryState> {
         value: (x: string, y?: string) => (y ? y : x),
         default: () => '',
       },
+      status: {
+        value: (x: NodeStatus | null, y?: NodeStatus | null) => (y ? y : x),
+        default: () => null,
+      },
     };
     super(graphState);
     this.functions = functions;
     this.database = database;
     this.companyName = companyName;
+    this.outputHandler = outputHandler;
+  }
+
+  private async handleStatusUpdate(status: NodeStatus) {
```

Comment:
If this is used by other subgraphs, it should be in utils
I think we can drop it the same place as NodeStatus (IMO is more like a NodeResponse but that's up to you)

Reply:
It's only used by this agent. Can move it if others will need it